zend_abstract_interface/hook/hook.c

```diff
@@ -124,6 +144,15 @@ static void zend_hash_iterators_remove(HashTable *ht) {
     }
 }
 
+
+static void zai_hook_static_inheritors_destroy(zval *zv) {
+    free(Z_PTR_P(zv));
```

Comment:
question: I have not found where these pointers are allocated, but AFAIK this function is to free the elements in `zai_hook_static_inheritors` (as the call to `zend_hash_init` suggests).
Long story short, is `free(Z_PTR_P(zv));` correct or should it be `pefree(Z_PTR_P(zv),1);`?
I understood, that `pefree(void *, 1)` is just a call to `free(void *)` but if my thought is right, this would be more consistent.

Is this the line corresponding to this `free()`?
https://github.com/DataDog/dd-trace-php/pull/1914/files#diff-f4b7c344c209da085fdb03ad5b08198ef92790f7fc43751e23486fc56a1bc857R446

Reply:
I think in my code I've nearly always been using free() - only pefree() when it's dynamic.

The distinction actually only matters for malloc/pemalloc(), as malloc is faillible, while pemalloc(, 1) is not.